packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -298,26 +299,35 @@ Plugin _pluginFromPackage(String name, Uri packageRoot) {
   );
 }
 
-List<Plugin> findPlugins(FlutterProject project) {
+Future<List<Plugin>> findPlugins(FlutterProject project) async {
   final List<Plugin> plugins = <Plugin>[];
-  Map<String, Uri> packages;
+  PackageConfig packageConfig;
   try {
     final String packagesFile = globals.fs.path.join(
       project.directory.path,
-      PackageMap.globalPackagesPath,
+      globalPackagesPath,
+    );
+    packageConfig = await loadPackageConfigUri(
```

Comment:
Oh. It looks like this function catches all exceptions coming from `loader`:

https://github.com/dart-lang/package_config/blob/master/lib/src/package_config_io.dart#L73

Should we be passing an `onError` to distinguish from an empty file?